gui/src/3D/controls/slider3D.ts

```diff
@@ -0,0 +1,246 @@
+    public get value(): number {
+        return this._value;
+    }
+
+    public set value(value: number) {
+        if (this._value === value) {
+            return;
+        }
+
+        this._value = Math.max(Math.min(value, this._maximum), this._minimum);
+        this.onValueChangedObservable.notifyObservers(this._value);
+    }
+
+    protected get start(): number {
+        if (!this.node) {
+            return -SLIDER_SCALING / 2;
+        }
+
+        return this._sliderBar.position.x - this._sliderBar.scaling.y / 2;
+    }
+
+    protected get end(): number {
+        if (!this.node) {
+            return SLIDER_SCALING / 2;
+        }
+
+        return this._sliderBar.position.x + this._sliderBar.scaling.y / 2;
+    }
+
+    protected get roundingToDecimal(): boolean {
+        if (!this.mesh) {
+            return false;
+        }
+
+        return this.minimum % 1 !== 0 || this.maximum % 1 !== 0 || this.step !== 0;
+    }
+    /**
+     * Gets the slider bar material used by this control
+     */
+    public get sliderBarMaterial(): StandardMaterial {
+        return this._sliderBarMaterial;
+    }
+
+    /**
+     * Gets the slider thumb material used by this control
+     */
+    public get sliderThumbMaterial(): StandardMaterial {
+        return this._sliderThumbMaterial;
+    }
+
+    // Mesh association
+    protected _createNode(scene: Scene): TransformNode {
+        const anchor = new TransformNode(`${this.name}_slider`, scene);
+
+        const sliderBar = MeshBuilder.CreateCylinder(`${this.name}_sliderbar`, { diameter: 0.03, height: 1.0 }, scene);
+        sliderBar.rotation.z = -Math.PI / 2;
+        sliderBar.scaling.y = SLIDER_SCALING;
+        sliderBar.isPickable = false;
+        sliderBar.setParent(anchor);
+
+        const sliderThumb = MeshBuilder.CreateBox(`${this.name}_sliderthumb`, { size: 0.1 }, scene);
+        sliderThumb.scaling = new Vector3(SLIDER_SCALING, SLIDER_SCALING, SLIDER_SCALING);
+        sliderThumb.position.x = this._convertToPosition(this.value);
+        sliderThumb.addBehavior(this._createBehavior());
+        sliderThumb.setParent(anchor);
+
+        this._sliderBar = sliderBar;
+        this._sliderThumb = sliderThumb;
+
+        return anchor;
+    }
+
+    protected _affectMaterial(mesh: AbstractMesh) {
+        const barMaterial = new StandardMaterial(`${this.name}_sliderbar_material`, mesh.getScene());
+        barMaterial.specularColor = Color3.Black();
+        this._sliderBar.material = barMaterial;
+
+        const thumbMaterial = new StandardMaterial(`${this.name}_sliderthumb_material`, mesh.getScene());
+        thumbMaterial.specularColor = Color3.Black();
+        mesh.material = thumbMaterial;
+
+        this._sliderBarMaterial = barMaterial;
+        this._sliderThumbMaterial = thumbMaterial;
+    }
+
+    private _createBehavior(): PointerDragBehavior {
+        const pointerDragBehavior = new PointerDragBehavior({ dragAxis: new Vector3(1, 0, 0) });

```

Comment:
>new Vector3(1, 0, 0) [](start = 72, length = 20)

nit: fwiw, some of these more common constants have static values. Like this vector3 here exists as a static Vector3.Right() value.

Reply:
Nice, I did not know there is values for this. I only know Vector3.Zero().